src/e2e-test/java/io/cdap/plugin/utils/SalesforceClient.java

```diff
@@ -266,4 +266,48 @@ public static String queryObjectId(String objectName) {
     }
     return uniqueRecordId;
   }
+
+    public static void updateObject(String id, String objectName) {
+    getAccessToken();
+    HttpClient httpClient = HttpClientBuilder.create().build();
+    String baseUri = loginInstanceUrl + REST_ENDPOINT + API_VERSION;
+    String uri = baseUri + "/sobjects/" + objectName + "/" + id;
+    HttpPatch httpPatch = new HttpPatch(uri);
+    Header oauthHeader = new BasicHeader("Authorization", "Bearer " + loginAccessToken);
+    httpPatch.addHeader(oauthHeader);
+    httpPatch.addHeader(prettyPrintHeader);
+
+    try {
+      String uniqueId = RandomStringUtils.randomAlphanumeric(7);
+      String newName = "testname" + uniqueId;
+      String newEmail = "test@gmail.com" + uniqueId;
+
+      // Create a map to hold the fields and values to update
+      Map<String, Object> fieldsToUpdate = new HashMap<>();
+      fieldsToUpdate.put("Name", newName);
+      fieldsToUpdate.put("Col_Email__c", newEmail);
+
+      // Convert the fieldsToUpdate map to a JSON string
+      Gson gson = new Gson();
+      String fieldsJson = gson.toJson(fieldsToUpdate);
+
+      // Set the request body with the fields to update
+      StringEntity requestEntity = new StringEntity(fieldsJson);
+      requestEntity.setContentType(ContentType.APPLICATION_JSON.getMimeType()); // Set content type to application/json
+      httpPatch.setEntity(requestEntity);
+
+      // Execute the update request
+      HttpResponse response = httpClient.execute(httpPatch);
+      int statusCode = response.getStatusLine().getStatusCode();
+
+      if (statusCode == 204) {
+        logger.info("Object updated Successfully");
+      } else {
+        String responseString = EntityUtils.toString(response.getEntity());
+        logger.info("Failed to update object. Response: " + responseString);
```

Comment:
we should fail the test if there's an error or unexpected exception thrown

Reply:
done